### PR TITLE
Improve readability of documenting_mpl.rst

### PR DESCRIPTION
**Pull request by SidharthBansal**
## PR Summary

Felt the sentence too long and confusing. Tried to make it more understandable.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: doc/devel/documenting_mpl.rst [hunk 1]
```diff
@@ -290,8 +290,8 @@ Other packages can also be linked via
 
 will return this link: `numpy.mean`.  This works for Python, Numpy, Scipy,
 and Pandas (full list is in :file:`doc/conf.py`). Sometimes it is tricky
-to get external Sphinx linking to work; to
-check that a something exists to link to the following shell command outputs
```
Comment (anntzer) [inline doc/devel/documenting_mpl.rst:294]: I agree the previous wording was less than optimal.  Perhaps

> This works for x, y, and z (full list is in ...).  If external linking fails, you can check the full list of referenceable objects with the following command::
> ...
> (adjust the url accordingly for projects other than numpy).

File: doc/devel/documenting_mpl.rst [hunk 2]
```diff
@@ -289,12 +289,14 @@ Other packages can also be linked via
...
 will return this link: `numpy.mean`.  This works for Python, Numpy, Scipy,
-and Pandas (full list is in :file:`doc/conf.py`). Sometimes it is tricky
-to get external Sphinx linking to work; to
-check that a something exists to link to the following shell command outputs
-a list of all objects that can be referenced (in this case for Numpy)::
+and Pandas (full list is in :file:`doc/conf.py`).  If external linking fails,
+you can check the full list of referenceable objects with the following
+commands::
 
+  python -m sphinx.ext.intersphinx 'https://docs.python.org/3/objects.in'
```
Comment (anntzer) [inline doc/devel/documenting_mpl.rst:296]: inv (typo)

Comment (SidharthBansal) [reply doc/devel/documenting_mpl.rst]: Sorry. Will pay more attention from next time.

**Comment by SidharthBansal** [comment]
Thanks all!

**Pull request merged**